src/QueryPipeline/RemoteQueryExecutor.cpp

```diff
@@ -125,6 +126,9 @@ RemoteQueryExecutor::RemoteQueryExecutor(
                 LOG_DEBUG(log, "Failed to connect to replica {}. {}", pool->getAddress(), fail_message);
             else
                 LOG_DEBUG(log, "Replica is not usable for remote query execution: {}. {}", pool->getAddress(), fail_message);
+
+            if (connection_pool_with_failover_)
+                connection_pool_with_failover_->incrementErrorCount(pool);
```

Comment:
I think we shouldn't increment the error count in case of the connection is not usable for a particular query (because of missing table), - it can be much usable for other queries.

Reply:
Indeed, I overlooked that point.